src/planet-a/IcoSphereTree.h

```diff
@@ -67,8 +81,70 @@ static constexpr int gc_icosahedronVertCount = 12;
 static constexpr std::uint8_t gc_triangleMaskSubdivided = 0b0001;
 //static constexpr std::uint8_t gc_triangleMaskChunked    = 0b0010;
 
-// An icosahedron with subdividable faces
-// it starts with 20 triangles, and each face can be subdivided into 4 more
+/**
+ * A simple 1D translation and scale
+ */
+struct TriangleSideTransform
+{
+    float m_translation{0.0f};
+    float m_scale{1.0f};
+};
+
+class IcoSphereTreeObserver
+{
+public:
+    using Handle_T = std::list<std::weak_ptr<IcoSphereTreeObserver>>::iterator;
+
+    virtual void on_ico_triangles_added(std::vector<trindex_t> const&) = 0;
+    virtual void on_ico_triangles_removed(std::vector<trindex_t> const&) = 0;
+    virtual void on_ico_vertex_removed(std::vector<vrindex_t> const&) = 0;
+};
+
+
+/**
+ * A triangle on the IcoSphereTree
+ */
+struct SubTriangle
+{
+    trindex_t m_parent;
+    uint8_t m_siblingIndex; // 0:Top 1:Left 2:Right 3:Center
+
+    trindex_t m_neighbours[3];
+    buindex_t m_corners[3]; // to vertex buffer, 3 corners of triangle
+
+    osp::Vector3 m_center;
+
+    bool m_subdivided{false}, m_deleted{false};
```

Comment:
Personally I'm not a fan of this syntax for declaring variables.

If you like it though, go ahead